resim/msg/converter_plugin.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include "resim/msg/converter_plugin.hh"
+
+#include <dlfcn.h>
+
+#include <utility>
+
+#include "resim/assert/assert.hh"
+
+namespace resim::msg {
+
+ConverterPlugin::ConverterPlugin(const std::filesystem::path &plugin_path)
+    : handle_(dlopen(plugin_path.c_str(), RTLD_LAZY)) {
+  REASSERT(handle_ != nullptr, "Failed to load plugin!");
```

Comment:
It might be useful to report the `dlerror()` value here.

Reply:
I've done so in the latest version!